fdbserver/BackupWorker.actor.cpp

```diff
@@ -508,6 +510,7 @@ ACTOR Future<Void> uploadData(BackupData* self) {
 		} else {
 			state int numMsg = 0;
 			for (const auto& message : self->messages) {
+				// Q: Why would message.getVersion() > maxPopVersion?
```

Comment:
1) peek is aggressive. prefetch version in peek. You may peek message that has not been committed. You cannot save data that has not been committed.
2)

Reply:
You gonna turn these lines into actual inline docs inside the code?